cpp/src/community/k_truss_impl.cuh

```diff
@@ -114,246 +188,547 @@ k_truss(raft::handle_t const& handle,
+
+      transform_e(
+        handle,
+        cur_graph_view,
+        edge_src_in_k_minus_1_cores.view(),
+        edge_dst_in_k_minus_1_cores.view(),
+        edge_dummy_property_t{}.view(),
+        [] __device__(auto, auto, auto src_in_k_minus_1_core, auto dst_in_k_minus_1_core, auto) {
+          return src_in_k_minus_1_core && dst_in_k_minus_1_core;
+        },
+        in_k_minus_1_core_edge_mask.mutable_view());
+
+      edge_mask = std::move(in_k_minus_1_core_edge_mask);
+      if (cur_graph_view.has_edge_mask()) { cur_graph_view.clear_edge_mask(); }
+      cur_graph_view.attach_edge_mask(edge_mask.view());
+    }
   }
 
   // 3. Keep only the edges from a low-degree vertex to a high-degree vertex.
 
-  {
-    auto cur_graph_view = modified_graph_view ? *modified_graph_view : graph_view;
-
-    auto vertex_partition_range_lasts =
-      renumber_map
-        ? std::make_optional<std::vector<vertex_t>>(cur_graph_view.vertex_partition_range_lasts())
-        : std::nullopt;
+  edge_src_property_t<decltype(cur_graph_view), edge_t> edge_src_out_degrees(handle,
+                                                                             cur_graph_view);
+  edge_dst_property_t<decltype(cur_graph_view), edge_t> edge_dst_out_degrees(handle,
+                                                                             cur_graph_view);
 
+  cugraph::edge_property_t<graph_view_t<vertex_t, edge_t, false, multi_gpu>, bool> dodg_mask(
+    handle, cur_graph_view);
+  {
     auto out_degrees = cur_graph_view.compute_out_degrees(handle);
-    edge_src_property_t<decltype(cur_graph_view), edge_t> edge_src_out_degrees(handle,
-                                                                               cur_graph_view);
-    edge_dst_property_t<decltype(cur_graph_view), edge_t> edge_dst_out_degrees(handle,
-                                                                               cur_graph_view);
     update_edge_src_property(
       handle, cur_graph_view, out_degrees.begin(), edge_src_out_degrees.mutable_view());
     update_edge_dst_property(
       handle, cur_graph_view, out_degrees.begin(), edge_dst_out_degrees.mutable_view());
 
-    rmm::device_uvector<vertex_t> srcs(0, handle.get_stream());
-    rmm::device_uvector<vertex_t> dsts(0, handle.get_stream());
+    cugraph::fill_edge_property(
+      handle, unmasked_cur_graph_view, dodg_mask.mutable_view(), bool{false});
+
+    cugraph::transform_e(
+      handle,
+      cur_graph_view,
+      edge_src_out_degrees.view(),
+      edge_dst_out_degrees.view(),
+      edge_dummy_property_t{}.view(),
+      [] __device__(auto src, auto dst, auto src_out_degree, auto dst_out_degree, auto) {
+        return (src_out_degree < dst_out_degree) ? true
+               : ((src_out_degree == dst_out_degree) &&
+                  (src < dst) /* tie-breaking using vertex ID */)
+                 ? true
+                 : false;
+      },
+      dodg_mask.mutable_view(),
+      false);
```

Comment:
```
dodg_mask.mutable_view());
```
No need to explicitly set do_expensive_check to false here.

Reply:
Right. But since we have the expensive check exposed, I can just pass it to every function that support it/